docs/userguide.md

```diff
@@ -0,0 +1,151 @@
+interact with it using kubectl:
+
+```sh
+$ kubectl get clusters --context <host_cluster_context>
+No resources found
+$
+```
+
+### Try it out!
+
+In these examples, `context` is either the `cluster_registry_context` if you did
+a standalone deployment, or the `host_cluster_context` if you did an aggregated
+deployment.
+
+Try creating a cluster:
+
+```sh
+kubectl apply -f - --context <context> <<EOF
+kind: Cluster
+apiVersion: clusterregistry.k8s.io/v1alpha1
+metadata:
+  name: test-cluster
+spec:
+  kubernetesApiEndpoints:
+    serverEndpoints:
+      - clientCidr: "0.0.0.0/0"
+        serverAddress: "100.0.0.0"
+EOF
+```
+
+And then reading it back:
+
+```sh
+kubectl get clusters --context <context>
+```
+
+### Advanced deployments
+
+`crinit` takes care of a lot of the plumbing work to get a cluster registry
+running, but the cluster registry can be run anywhere you like, as long as you
+configure it correctly. In fact, we have an [example]() that runs the cluster
+registry as an aggregated API server in a docker container
+
+#### API server
+Since the cluster registry is an API server, all of the guidance around
+deploying and managing aggregated API servers applies.
+[The Kubernetes docs](https://kubernetes.io/docs/concepts/api-extension/apiserver-aggregation/)
+have some pointers about aggregation. You can also look at the
+[API Server Concepts docs](https://github.com/kubernetes-incubator/apiserver-builder/tree/master/docs/concepts)
+for some more technical guidance about aggregating. If you wish to run the
+cluster registry as a standalone server, you will need to manage the
+certificates and client authentication/authorization yourself. You can use any
+authentication mode supported by the `clusterregistry` command, but you will
+probably need to use an
+[authorizing webhook](https://kubernetes.io/docs/admin/authorization/webhook/),
+since the cluster registry does not provide RBAC support inherently.
+
+#### etcd
+
+`crinit` by default deploys `etcd` in a container in the same pod as the
+`clusterregistry` binary. This makes it easy to get started, but is not
+necessarily a good strategy for deploying a cluster registry for production use.
+You may want to look into the
+[etcd operator](https://github.com/coreos/etcd-operator) for production
+deployments.
+
+## Interacting with the cluster registry
+
+### kubectl
+
+The cluster registry is a Kubernetes API server, and you can interact with it
+using standard `kubectl` commands. It provides one API type, `clusters`, which
+you can create, get, list and delete like any other Kubernetes Object. See
+[Try it out!](#try-it-out) above for some sample commands.
+
+### Generated Go client
+
+There is a generated Go client library for the cluster registry in
+[/pkg/client](/pkg/client). You can vendor in the cluster registry repository
+and use the client library directly from your Go code.
+
+### OpenAPI spec
+
+There is an OpenAPI spec file provided [here](/api/swagger.json). You can use
```

Comment:
Is the URL legit? If it isn't I would just drop here and tell users how to get the spec from their API server.

Reply:
It is, surprisingly: GitHub takes absolute path links without schemes and references them from the root of the repository.

Reply:
Oh wow! Didn't know this. TIL.